### PR TITLE
feat(coverage): custom reporter support

### PR DESCRIPTION
**Pull request by AriPerkkio**
### Description

- Closes https://github.com/vitest-dev/vitest/issues/4150

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: test/coverage-test/test/configuration-options.test-d.ts [hunk 1]
```diff
@@ -209,12 +205,6 @@ test('reporters, with options', () => {
 
   assertType<Coverage>({
     reporter: [
-      // @ts-expect-error -- teamcity report option on html reporter
-      ['html', { blockName: 'string' }],
-
-      // @ts-expect-error -- html-spa report option on json reporter
-      ['json', { metricsToShow: ['branches'] }],
-
```
Comment (AriPerkkio) [inline test/coverage-test/test/configuration-options.test-d.ts:217]: I wasn't able to keep options types supported after adding support for `string` as reporter name. Built-in reporters' options are still prompted by intellisense though.

File: docs/guide/coverage.md [hunk 2]
```diff
@@ -70,6 +70,55 @@ export default defineConfig({
 })
 ```
 
+## Custom Coverage Reporter
+
+You can use custom coverage reporters by passing either name of the package or absolute path in `test.coverage.reporter`:
```
Comment (dammy001) [inline docs/guide/coverage.md:75]:
```suggestion
You can use custom coverage reporters by passing either the name of the package or absolute path in `test.coverage.reporter`:
```

File: docs/guide/coverage.md [hunk 3]
```diff
@@ -70,6 +70,55 @@ export default defineConfig({
...
+
+        // Specify reporter using local path
+        '/absolute/path/to/custom-reporter.cjs',
+      ],
+    },
+  },
+})
+```
+
+Custom reporters are loaded by Istanbul and must match its reporter interface. See [built-in reporters' implementation](https://github.com/istanbuljs/istanbuljs/tree/master/packages/istanbul-reports/lib) for reference.
```
Comment (dammy001) [inline docs/guide/coverage.md:96]: shouldn't this be wrapped in some sort of info card?

**Review by dammy001 — APPROVED** [review]

File: docs/guide/coverage.md (hunk 3 above)
Comment (AriPerkkio) [reply docs/guide/coverage.md:96]: As it's describing the code block below it I think it should be as is, and not wrapped in `:::tip` card.

**Review by sheremet-va — APPROVED** [review]

**Pull request merged**